### PR TITLE
Q8

### PR DESCRIPTION
**Pull request by Byteuth**
Adds CORE: PATCH /api/articles/:article_id
Adds tests for new PATCH
Adds new controllers, models functions
Adds testing for new models 

Adds jest --runInBand
updates endpoints.json

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -123,7 +123,44 @@ describe("/api/articles/:article_id", () => {
...
+        return request(app)
+        .patch("/api/articles/999")
+        .send(inputVote)
+        .expect(404)
+        .then((response ) => {
+            
+            expect(response.body.msg).toBe("Not Found");
+        });
+    });
+
```
Comment (MKD01) [inline __tests__/app.test.js:163]: This is a good start to error handling but there are a few that are required for this endpoint as well. 

What should happen when a request is made using an incorrect article id? I.e. /api/articles/banana
What should happen if the given inputVote is missing inc_votes/ is the wrong data type?

File: __tests__/utils.models.test.js [hunk 2]
```diff
@@ -86,5 +87,30 @@ describe("insertComment", () => {
...
+        await updateVotesValue(articleId, -100)
+        .then((result) => {
+            expect(result).toBeObject()
+            expect(result.votes).toBe(24)
+        })
+
+    })
+
+})
+
```
Comment (MKD01) [inline __tests__/utils.models.test.js:114]: This shouldnt be tested for as we are already doing this using the endpoint testing above

File: __tests__/app.test.js [hunk 3]
```diff
@@ -123,7 +123,44 @@ describe("/api/articles/:article_id", () => {
...
+        return request(app)
+        .patch("/api/articles/1")
+        .send(inputVote)
+        .expect(200)
+        .then((response ) => {
+            const updatedArticle = response.body.updatedArticle
+            expect(updatedArticle).toMatchObject(control)
+        });
+    });
+
```
[the remark this answers is not in the document]
Comment (Byteuth) [reply __tests__/app.test.js:151]: +

File: __tests__/app.test.js (hunk 1 above)
Comment (Byteuth) [reply __tests__/app.test.js:163]: Will address this,

File: __tests__/utils.models.test.js (hunk 2 above)
Comment (Byteuth) [reply __tests__/utils.models.test.js:114]: Might need some assistance with this, please advise

My thinking here is making so app.test PATCH checks if the return object matches the control result object on a request basis, while utils.models checks if each request and subsequent requests modify .votes to the correct value

would it be advisable to invoque updateVotesValue in app.test and test from there?

**Pull request merged**